[WIP] HTTP2 support (#832)
sindresorhus/got: readme.md

###### request

Type: `Function`<br>

> Default: `http.request` `https.request` `http2wrapper.request` *(Depending on the protocol)*

Reviewer: Can you linkify `http2wrapper.request` to your package?

> Custom request function. The main purpose of this is to [support HTTP2 using a wrapper](#http2-support).

Reviewer: > The main purpose of this is to [support HTTP2 using a wrapper](#http2-support).

This is misleading as it's no built-in. I would just drop that sentence or document a different use-case.

**Tip:** Need to merge some instances into a single one? Check out [`got.mergeInstances()`](advanced-creation.md#merging-instances).

### HTTP2 support

Got supports HTTP2 via the [`http2-wrapper`](https://github.com/szmarczak/http2-wrapper) package.<br>

> To make this work, you need to use Node v10 or higher.

Reviewer: ```suggestion
To make this work, you need to use Node.js 10 or higher.
```

> **Note:** using custom `request` function will disable HTTP2 support.

Reviewer: ```suggestion
**Note:** Using the `request` option will disable HTTP2 support.
```

```js
const got = require('got');

(async () => {
	const {headers} = await got('https://nghttp2.org/httpbin/anything');

> 	console.log(headers.via); // => 2 nghttpx

Reviewer: ```suggestion
	console.log(headers.via);
	//=> '2 nghttpx'
```
